### PR TITLE
[WIP] Perfomance test with multiple tabs

### PR DESCRIPTION
**Pull request by jagadeshjai**
-  Add a new benchmark to open multiple tabs for the given stories.

Resolves https://github.com/brave/brave-browser/issues/40515

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: tools/perf/brave_page_sets/brave_loading_desktop_pages.py [hunk 1]
```diff
@@ -69,16 +65,69 @@ def AddStories(self, tags, urls, startup_delay=False):
...
+  class MultiTabStory(page_cycler_story.PageCyclerStory):
+    """ A story that opens multiple tabs. """
+    TAB_COUNT = 10
+
+    def RunPageInteractions(self, action_runner):
+      tabs = action_runner.tab.browser.tabs
+      tabs[0].Navigate(self.url)
+      for _ in range(1, self.TAB_COUNT):
+        new_tab = tabs.New()
+        new_tab.action_runner.Navigate(self.url)
```
Comment (atuchin-m) [inline tools/perf/brave_page_sets/brave_loading_desktop_pages.py:123]: This part looks good, it actually opens n-tabs.
The main questions are:
1) what metrics do we get
2) from all the tabs or from just one
3) how we aggregate them

Comment (atuchin-m) [reply tools/perf/brave_page_sets/brave_loading_desktop_pages.py:123]: The main metric we are interested in is the total memory usage (private_footprint)

File: tools/perf/brave_page_sets/brave_loading_desktop_pages.py [hunk 2]
```diff
@@ -33,26 +33,22 @@ def _StartBrowser(self, page):
     time.sleep(10)
 
 
-class BraveLoadingDesktopStorySet(story.StorySet):
-  """ Brave version of LoadingDesktopStorySet.
+class BaseBraveLoadingDesktopStorySet(story.StorySet):
```
Comment (atuchin-m) [inline tools/perf/brave_page_sets/brave_loading_desktop_pages.py:36]: BTW, the original chromium benchmark `loading.desktop` was discounted sometime ago.
One day we could also drop `loading.desktop.brave` in favor of `system_health.*`

File: tools/perf/brave_page_sets/brave_loading_desktop_pages.py (hunk 1 above)
Comment (jagadeshjai) [reply tools/perf/brave_page_sets/brave_loading_desktop_pages.py:123]: Yes, I'll handle it using system_health.memory*, which will ease our implementation for the requirement.

File: tools/perf/brave_page_sets/brave_loading_desktop_pages.py (hunk 2 above)
Comment (jagadeshjai) [reply tools/perf/brave_page_sets/brave_loading_desktop_pages.py:36]: I have created another [PR](https://github.com/brave/brave-core/pull/25193) with `system_health...`, So probably I can close this PR.

**Comment by jagadeshjai** [comment]
Closing this PR as the intended changes have been made [here](https://github.com/brave/brave-core/pull/25255).

**Pull request closed without merging**